### DISCUSSION TITLE
How to get count of distinct records from MongoDB, through STRAPI

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7440] [Legacy Topics]
I am trying to find count of distinct records from a collection, MongoDB acts as datastore. What would be sample snippet that i can add to a controller to achieve this?

In fact, i need to perform same operation on 4 different collections and return them in single response. Any pointers would be of great help! As i am not getting much of information from mongoosejs website.

**Comment by derrickmehaffy** [comment]
Hi @basavarajdodamani you are indeed on the right path here, in this case a custom route/controller would be needed. You will want to check on the [custom queries](https://strapi.io/documentation/v3.x/concepts/queries.html#custom-queries) documentation and refer to the mongoose API docs to find the correct method to select distinct values.

Handling multiple queries is no problem within a controller and you can customize the response fairly easily. Let me know if you have specific questions!

**Reply by basavarajdodamani** [reply]
@derrickmehaffy Thanks for the response. Yes somehow i cant get it working.
I have a collection called ticket and ConsumerID as its one of the field. I would like to get the distinct count of ConsumerID from the controller. I cant get this working as it keeps throwing errors such as ticket or find or distinct is not defined. 

dCount: async ctx => {
		let ticket;
	console.log('inside controller count' );
	ticket.find().distinct('ConsumerID').exec(callback);
	//strapi.query(ticket, plugin).model;
	 },

Could you please suggest.

**Reply by basavarajdodamani** [reply]
I have got it working using strapi query which looks as below:
const result = strapi.query('ticket').model.find().distinct('ConsumerID').exec(function (err, count)
